src/System.Globalization/tests/CharUnicodeInfo/CharUnicodeInfoTests.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Globalization.Tests
+{
+    public class CharUnicodeInfoGetUnicodeCategoryTests
+    {
+        [Fact]
+        public void GetUnicodeCategory()
+        {
+            foreach (CharUnicodeInfoTestCase testCase in CharUnicodeInfoTestData.TestCases)
+            {
+                if (testCase.Utf32CodeValue.Length == 1)
+                {
+                    // Test the char overload for a single char
+                    GetUnicodeCategory(testCase.Utf32CodeValue[0], testCase.GeneralCategory);
+                }
+                else
+                {
+                    // Test the string overload for a surrogate pair
+                    GetUnicodeCategory(testCase.Utf32CodeValue, 0, testCase.GeneralCategory);
+                }
+            }
+        }
+        
+        [Theory]
+        [InlineData('\uFFFF', UnicodeCategory.OtherNotAssigned)]
```

Comment:
I searched the Unicode.8.0.txt dataset but didn't see FFFF
I think that's because the dataset only contains assigned chars